src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/DocumentsController.kt

```diff
@@ -38,14 +39,16 @@ class DocumentsController(
       is AuthorisableActionResult.Success -> documentsMetaDataResult.entity
     }
 
-    if (! documentsMetaData.documentExists(documentId)) {
-      throw NotFoundProblem(documentId, "Document")
-    }
+    val documentMetaData = documentsMetaData.findDocument(documentId)
+      ?: throw NotFoundProblem(documentId, "Document")
 
     return ResponseEntity(
       StreamingResponseBody { outputStream ->
         offenderService.getDocument(crn, documentId, outputStream)
       },
+      HttpHeaders().apply {
+        put("Content-Disposition", listOf("attachment; filename=\"${documentMetaData.documentName}\""))
```

Comment:
Why is this a `listOf`?

Reply:
Any HTTP header that can be represented as a comma seperated list can also be sent as multiple key-value pairs, e.g.

```
Cache-Control: no-cache, no-store
```

is the same as:

```
Cache-Control: no-cache
Cache-Control: no-store
```

So Spring types HttpHeaders as a Multi-Value map, i.e. `Map<String, List<String>>` to support the second case